internal/secretsource/url/url.go

```diff
@@ -0,0 +1,256 @@
+	}
+
+	body, err := io.ReadAll(response.Body)
+	if err != nil {
+		return "", fmt.Errorf("failed to read response: %w", err)
+	}
+
+	// Extract secret value from response
+	secret, err := extractSecretFromResponse(body, us.config.ResponsePath)
+	if err != nil {
+		return "", fmt.Errorf("failed to extract secret: %w", err)
+	}
+
+	return secret, nil
+}
+
+type limiter interface {
+	Wait(ctx context.Context) error
+}
+
+func newLimiter(requestsPerMinuteLimit, requestsBurst int) *rate.Limiter {
+	// The calculation below looks wrong because it seems like it's giving
+	// n min²/req, but the first number is actually time unit/min, so the
+	// units of the result are time unit/req, which is correct because it's
+	// the interval of time after which a new token is replenished. In
+	// other words, the units are time unit/token.
+	tokenReplenishInterval := time.Minute / time.Duration(requestsPerMinuteLimit)
+
+	return rate.NewLimiter(rate.Every(tokenReplenishInterval), requestsBurst)
+}
+
+func extractSecretFromResponse(body []byte, responsePath string) (string, error) {
+	if responsePath == "" {
+		// If no path specified, assume response body is the secret
+		return string(body), nil
+	}
+
+	var data map[string]interface{}
+	if err := json.Unmarshal(body, &data); err != nil {
+		return "", fmt.Errorf("failed to decode JSON response: %w", err)
+	}
+
+	// Simple JSON path traversal for nested keys like "data.value"
+	parts := strings.Split(responsePath, ".")
+	current := interface{}(data)
+
+	for _, part := range parts {
+		if m, ok := current.(map[string]interface{}); ok {
+			val, exists := m[part]
+			if !exists {
+				return "", fmt.Errorf("path component %q not found in response", part)
+			}
+			current = val
+		} else {
+			return "", fmt.Errorf("cannot traverse path at %q: not a JSON object", part)
+		}
+	}
+
+	if secret, ok := current.(string); ok {
+		return secret, nil
+	}
+
+	return "", fmt.Errorf("secret value at path %q is not a string", responsePath)
+}
+
+func parseConfigArgument(configArg string) (string, error) {
+	configKey, configPath, ok := strings.Cut(configArg, "=")
+	if !ok || configKey != "config" {
+		return "", errInvalidConfig
+	}
+
+	return configPath, nil
+}
+
+func getConfig(arg string, fs fsext.Fs) (extConfig, error) {
+	var config extConfig
+
+	// Parse the ConfigArgument to get the config file path
+	configPath, err := parseConfigArgument(arg)
+	if err != nil {
+		return config, err
+	}
+
+	file, err := fs.Open(configPath)
+	if err != nil {
+		return config, fmt.Errorf("failed to open config file: %w", err)
+	}
+	defer func() { _ = file.Close() }()
+
+	configData, err := io.ReadAll(file)
+	if err != nil {
+		return config, fmt.Errorf("failed to read config file: %w", err)
+	}
+
+	if err := json.Unmarshal(configData, &config); err != nil {
+		return config, fmt.Errorf("failed to parse JSON config: %w", err)
+	}
+
+	if config.URLTemplate == "" {
+		return config, errMissingURLTemplate
+	}
```

Comment:
I suggest to also validate that the URL template is a valid url too. I would also validate that the URL scheme is https.